tools/dnt_preset_editor.html

```diff
@@ -1139,6 +1338,695 @@
+
+        for (let i = 0; i < maxDisplayParams; i++) {
+            let param = globalParamList[i];
+            let nickname = param.nickname || param.name.toLowerCase().replace(/\s+/g, '_');
+            let colIndex = i % 3;
+            let rowIndex = Math.floor(i / 3);
+            let cX = colPositions[colIndex];
+            let rY = rowPositions[rowIndex];
+
+            // Determine what text to display
+            let displayLine;
+            let paramIndexForHighlight = param.paramIndex; // used for highlight
+            if (param.allControlOptions && param.allControlOptions.length > 0) {
+                drawCalls.push(`    drawText(${cX}, ${rY}, string.format("%s: %s", "${param.nickname.toUpperCase()}", getEnumText(${nickname}_options, ${nickname}_value)))`);
+                displayLine = `${param.nickname.toUpperCase()}: [Enum]`;
+            } else {
+                if (param.unit === 'dB') {
+                    drawCalls.push(`    local ${nickname}_db = to_db(${nickname}_value or 0)`);
+                    drawCalls.push(`    drawText(${cX}, ${rY}, string.format("%s: %3.1fdB", "${param.nickname.toUpperCase()}", ${nickname}_db))`);
+                    displayLine = `${param.nickname.toUpperCase()}: dB value`;
+                } else {
+                    drawCalls.push(`    local ${nickname}_percent = (${nickname}_value or 0) * 100`);
+                    drawCalls.push(`    drawText(${cX}, ${rY}, string.format("%s: %3.1f", "${param.nickname.toUpperCase()}", ${nickname}_percent))`);
+                    displayLine = `${param.nickname.toUpperCase()}: % value`;
+                }
+            }
+
+            // add to drawCommands for preview
+            let cXVal = [10, 74, 138][colIndex];
+            let rYVal;
+            switch (rowIndex) {
+                case 0: rYVal = 20; break;
+                case 1: rYVal = 35; break;
+                case 2: rYVal = 50; break;
+                case 3: rYVal = 65; break;
+            }
+
+            drawCommands.push({
+                type: 'text',
+                x: cXVal,
+                y: rYVal,
+                text: displayLine,
+                shade: 15,
+                paramIndex: paramIndexForHighlight
+            });
+        }
+
+        if (globalParamList.length > 12) {
+            drawCalls.push(`    -- Note: More than 12 parameters are mapped but only 12 are displayed.`);
+            drawCommands.push({
+                type: 'text',
+                x: 10,
+                y: 60,
+                text: "Only 12 displayed",
+                shade: 8
+            });
+        }
+
+        // Build final Lua script
+        let finalScript = [];
+        finalScript.push('return {');
+        finalScript.push(`    name = '${presetName} UI script',`);
+        finalScript.push(`    author = 'chordsmaze',`);
+        finalScript.push(`    description = 'UI script for ${presetName} preset created by the UI Builder by chordsmaze',`);
```

Comment:
If this is going in the Expert Sleeepers repo I would prefer it not to say "by chordsmaze" here.

Reply:
Sure I can edit